### ISSUE TITLE
suspicious_else_formatting warns when there's commented code

### ISSUE THREAD

**Issue by antoyo** [rust-lang/rust-clippy#12497]
### Summary

The lint `suspicious_else_formatting` warn when there's a commented `else if` between a `if/else`.
For instance, the following code:
```rust
fn main() {
    if true {
        println!("true");
    }
    /*else if false {
    }*/
    else {
        println!("false");
    }
}
```
will trigger:
```
error: this is an `else {..}` but the formatting might hide it
 --> src/main.rs:4:6
  |
4 |       }
  |  ______^
5 | |     /*else if false {
6 | |     }*/
7 | |     else {
  | |_________^
  |
  = note: to remove this lint, remove the `else` or remove the new line between `else` and `{..}`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#suspicious_else_formatting
  = note: `-D clippy::suspicious-else-formatting` implied by `-D warnings`
  = help: to override `-D warnings` add `#[allow(clippy::suspicious_else_formatting)]`
```
Now, I could format the code like this to please clippy:
```rust
fn main() {
    if true {
        println!("true");
    }
    /*else if false {
    }*/ else {
        println!("false");
    }
}
```
However, `rustfmt` will reformat the code so that the `else` is on a newline, which will trigger the clippy warning again.
So, there's a conflict between `clippy` and `rustfmt`.
I suggest `clippy` should not warn when there's a comment before the `else`.

### Lint Name

suspicious_else_formatting

### Reproducer

I tried this code:

```rust
fn main() {
    if true {
        println!("true");
    }
    /*else if false {
    }*/
    else {
        println!("false");
    }
}
```

I saw this happen:

```
error: this is an `else {..}` but the formatting might hide it
 --> src/main.rs:4:6
  |
4 |       }
  |  ______^
5 | |     /*else if false {
6 | |     }*/
7 | |     else {
  | |_________^
  |
  = note: to remove this lint, remove the `else` or remove the new line between `else` and `{..}`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#suspicious_else_formatting
  = note: `-D clippy::suspicious-else-formatting` implied by `-D warnings`
  = help: to override `-D warnings` add `#[allow(clippy::suspicious_else_formatting)]`
```

I expected to see this happen: no warning from clippy.

### Version

```text
rustc 1.78.0-nightly (2d24fe591 2024-03-09)
binary: rustc
commit-hash: 2d24fe591f30386d6d5fc2bb941c78d7266bf10f
commit-date: 2024-03-09
host: x86_64-unknown-linux-gnu
release: 1.78.0-nightly
LLVM version: 18.1.0
```
